components/formats-bsd/src/loci/formats/in/OMETiffReader.java

```diff
@@ -964,45 +968,67 @@ else if (t == 0) {
 
       // verify that all planes are available
       LOGGER.debug("    --------------------------------");
+      int validPlanes = num;
       for (int no=0; no<num; no++) {
         LOGGER.debug("    Plane[{}]: file={}, IFD={}",
           new Object[] {no, planes[no].id, planes[no].ifd});
         if (planes[no].reader == null) {
-          LOGGER.warn("Image ID '{}': missing plane #{}.  " +
-            "Using TiffReader to determine the number of planes.",
+          validPlanes--;
+          LOGGER.warn("Image ID '{}': missing plane #{}",
             meta.getImageID(i), no);
-          TiffReader r = new TiffReader();
-          r.setId(currentId);
-          try {
-            planes = new OMETiffPlane[r.getImageCount()];
-            for (int plane=0; plane<planes.length; plane++) {
-              planes[plane] = new OMETiffPlane();
-              planes[plane].id = currentId;
-              planes[plane].reader = r;
-              planes[plane].ifd = plane;
-            }
-            num = planes.length;
-          }
-          finally {
-            r.close();
+        }
+      }
+      // only use the current file's IFD count if this is a single file set
+      // otherwise, we risk pulling in IFDs from a different Image
+      if (validPlanes < num && files.size() <= 1) {
+        LOGGER.warn("Using TiffReader to determine the number of planes.");
+        TiffReader r = new TiffReader();
+        r.setId(currentId);
+        try {
+          planes = new OMETiffPlane[r.getImageCount()];
+          for (int plane=0; plane<planes.length; plane++) {
+            planes[plane] = new OMETiffPlane();
+            planes[plane].id = currentId;
+            planes[plane].reader = r;
+            planes[plane].ifd = plane;
           }
+          num = planes.length;
+        }
```

Comment:
Might be helpful to rename `num` to `planeCount` for clarity.

Also, this fallback (including before this PR) doesn't cater for the case where the plane count in the TIFF file is less than the image count dictated by the OME metadata.  In this case, we're liable to get an ArrayOutOfBounds exception when we try to access a valid plane index outside the `planes` sizes.  Maybe if `num != r.getImageCount()` we should bail out properly.  And also bail when the series count is greater than 1, because we have no hope of mapping higher series to the correct IFD.  At some level, when the TiffData is invalid or missing, we should not try to work around the brokenness, but error properly.